server/channels/store/sqlstore/access_control_policy_store.go

```diff
@@ -0,0 +1,404 @@
+	}
+
+	return nil
+}
+
+func (s *SqlAccessControlPolicyStore) SetActiveStatus(rctx request.CTX, id string, active bool) (*model.AccessControlPolicy, error) {
+	tx, err := s.GetMaster().Beginx()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to start transaction")
+	}
+	defer finalizeTransactionX(tx, &err)
+
+	existingPolicy, err := s.getT(rctx, tx, id)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to fetch policy with id=%s", id)
+	} else if errors.Is(err, sql.ErrNoRows) {
+		return nil, store.NewErrNotFound("AccessControlPolicy", id)
+	}
+
+	// also make sure if the policy is valid before updating active status
+	// just in case
+	existingPolicy.Active = active
+	if appErr := existingPolicy.IsValid(); err != nil {
+		return nil, appErr
+	}
+
+	tmp, err := fromModel(existingPolicy)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to parse policy with id=%s", id)
+	}
+
+	query := `Update AccessControlPolicies
+	SET Active = :Active WHERE ID = :ID`
+	_, err = tx.NamedExec(query, tmp)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to update policy with id=%s", id)
+	}
+
+	if err = tx.Commit(); err != nil {
+		return nil, errors.Wrap(err, "commit_transaction")
+	}
+
+	return existingPolicy, nil
+}
+
+func (s *SqlAccessControlPolicyStore) Get(_ request.CTX, id string) (*model.AccessControlPolicy, error) {
+	p := storeAccessControlPolicy{}
+	query := s.selectQueryBuilder.Where(sq.Eq{"ID": id})
+
+	err := s.GetReplica().GetBuilder(&p, query)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return nil, store.NewErrNotFound("AccessControlPolicy", id)
+		}
+		return nil, errors.Wrapf(err, "failed to find policy with id=%s", id)
+	}
+
+	policy, err := p.toModel()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to parse policy with id=%s", id)
+	}
+
+	return policy, nil
+}
+
+func (s *SqlAccessControlPolicyStore) getT(_ request.CTX, tx *sqlxTxWrapper, id string) (*model.AccessControlPolicy, error) {
+	query := s.getQueryBuilder().
+		Select(accessControlPolicySliceColumns()...).
+		From("AccessControlPolicies").
+		Where(
+			sq.Eq{"ID": id},
+		)
+
+	sql, args, err := query.ToSql()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to build query for policy with id=%s", id)
+	}
+
+	var storePolicy storeAccessControlPolicy
+	err = tx.Get(&storePolicy, sql, args...)
+	if err != nil {
+		return nil, err
+	}
+
+	policy, err := storePolicy.toModel()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to parse policy with id=%s", id)
+	}
+
+	return policy, nil
+}
+
+func (s *SqlAccessControlPolicyStore) GetAll(_ request.CTX, opts store.GetPolicyOptions) ([]*model.AccessControlPolicy, error) {
+	p := []storeAccessControlPolicy{}
+	query := s.selectQueryBuilder
+
+	if opts.ParentID != "" {
+		if s.DriverName() == model.DatabaseDriverPostgres {
+			query = query.Where(sq.Expr("Data->'imports' @> ?", fmt.Sprintf("[%q]", opts.ParentID)))
```

Comment:
Potential SQL injection site. Can we use placeholders here? I'm saying this because I had to convert a lot of similar queries to use placeholders and query builders as part of SQL injection prevention task a while ago.

Reply:
@harshilsharma63 there are already placeholders there, am I missing something?

Reply:
Got it, removed the `[]` part.